### ISSUE TITLE
v2.0 Roadmapping

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#3081]
- [x] Semver Auto update support
- [ ] Feature Flags
  - NOTES: Gutenberg and Woo have feature flags that we can draw inspiration from
- [x] Update GraphQL-PHP (to whatever version that drops support for PHP <7.4)
  - Probably need to do some communication around this in advance to let folks know this is coming
    - email the mailing list 
    - write a blog post
    - make use of `register_graphql_admin_notice()`
    - draft a policy about minimum supported versions of PHP and WordPress
- [ ] Remove existing deprecations
- [ ] Add new WPGraphQL IDE into core

**Comment by jasonbahl** [comment]
@justlevine as we're nearing the 2.0 release (ideally next week, as announced in the blog post: https://www.wpgraphql.com/2024/12/16/wpgraphql-v2-0-is-coming-heres-what-you-need-to-know) I wanted to follow-up on this. 

I think short-term we're still leaving the new IDE as a standalone plugin as it will likely see some iteration and own breaking change release cycles for some time, so I'm crossing that off the list. Possibly in the future it will be back in the core plugin, but for now we'll leave it out.

The last remaining items are: 

- [ ] feature flags (we have this PR: https://github.com/wp-graphql/wp-graphql/pull/3098)
- [ ] Remove existing deprecations

Curious if you had other ideas to try and work into this 2.0 and/or if we should even consider these for post-2.0 at this point.

**Comment by justlevine** [comment]
> [@justlevine](https://github.com/justlevine) as we're nearing the 2.0 release (ideally next week, as announced in the blog post: https://www.wpgraphql.com/2024/12/16/wpgraphql-v2-0-is-coming-heres-what-you-need-to-know) I wanted to follow-up on this [...]

Yeah that list went stale a while ago, not losing sleep over _these_ items.

> 
> Curious if you had other ideas to try and work into this 2.0

What I really want is an actual tagged v2-beta.1 (or even just merge https://github.com/wp-graphql/wp-graphql/pull/3247 onto the v2 parent branch), so we can actually test against (and have a clear way to patch and retest) v2 before it's released.

> Considering the v2.0 announcement and technical update guide did not mention the removal of deprecations (see: https://www.wpgraphql.com/2024/12/16/wpgraphql-v2-0-technical-update-breaking-changes), I'm thinking we follow up with the removal in the 3.0 release so we can properly communicate their removal in advance. 🤷‍♂

This is IMO the relevant part of the announcement:

>>By adopting the latest graphql-php, WPGraphQL inherits several breaking changes that may impact developers extending WPGraphQL. Below, we’ve highlighted the most important changes, though we also recommend reviewing the [graphql-php release notes](https://github.com/webonyx/graphql-php/releases) for more detail.

Or more specifically I'm concerned there are conflicting/incompatible changes to types that:

A. may affect deprecated functionality that we havn't planned to remove.
B. may force bad v2 patterns that we know we're going to need to remove immediately in a short-turnaround v3.
C. may cause a domino effect of A +B in v3 because we didn't proactively deprecate something in v2.

**Comment by jasonbahl** [comment]
Closing — this was the v2.0 roadmap and v2.0 has long shipped (currently 2.15.0). Dispositions:

- **Semver auto-update / graphql-php upgrade** — done, shipped in 2.0.
- **Feature flags** — shipped as the **Experiments API** (`src/Experimental/`, see the experiments docs).
- **New IDE into core** — intentionally kept standalone (per discussion above).
- **Remove existing deprecations** — moved to the v3.0 tracking issue #3902, where it's now inventoried.

@justlevine's concern about graphql-php-inherited type changes was tied to the pre-2.0 graphql-php upgrade, which has since shipped in 2.0 — so it's been overtaken by events. If any v2 pattern adopted from graphql-php still turns out to need removal, it can be folded into #3902.

**Comment by jasonbahl** [comment]
Picking one thread back up from here: the "draft a policy about minimum supported versions of PHP and WordPress" sub-item under the GraphQL-PHP bullet never got done, and it's newly relevant. Telemetry shows a meaningful number of sites still on PHP 7.4 (EOL since Nov 2022), so before we raise the floor we'd want exactly the advance communication this note called for.

I've opened #3925 to track it: publish the support policy (target the lowest PHP version still getting security support, following the php.net calendar), and ship a non-breaking heads-up first, an admin notice via `register_graphql_admin_notice()` plus a `graphql_debug` message, for sites below the target, so people get a runway before the floor actually moves in a major.
